packages/scrollregion/demo/stories/ScrollRegionStory.tsx

```diff
@@ -0,0 +1,87 @@
+/**
+ * Copyright Zendesk, Inc.
+ *
+ * Use of this source code is governed under the Apache License, Version 2.0
+ * found at http://www.apache.org/licenses/LICENSE-2.0.
+ */
+
+import React, { forwardRef, HTMLAttributes, RefObject } from 'react';
+import { Story } from '@storybook/react';
+import {
+  IScrollRegionContainerProps,
+  IUseScrollRegionProps,
+  ScrollRegionContainer,
+  useScrollRegion
+} from '@zendeskgarden/container-scrollregion';
+import classNames from 'classnames';
+
+interface IComponentProps extends HTMLAttributes<HTMLDivElement> {
+  height: number;
+  width: number;
+}
+
+const Component = forwardRef<HTMLElement, IComponentProps>(
+  ({ children, height, width, ...props }, ref) => (
+    <div
+      className={classNames(
+        'border',
+        'border-solid',
+        'overflow-scroll',
+        props.tabIndex === 0 ? 'border-blue-600' : 'border-grey-300'
+      )}
+      style={{ height, width }}
+      ref={ref as RefObject<HTMLDivElement>}
```

Comment:
Just out of curiosity, why do we do `forwardRef<HTMLElement, X>` versus forwardRef<HTMLDivElement, X>` and cast to a `RefObject<HTMLDivElement>` here?

Reply:
Yeah, this has a bad code smell. And it goes all the way back to the container API. I've updated the container interfaces with generics that take `HTMLElement` as a default, but can be overridden in order to specify the `containerRef` type (which, in the case of this story should be `HTMLDivElement`).

We should get better at using generics throughout Garden. 😉